torch/testing/_internal/common_methods_invocations.py

```diff
@@ -14948,19 +14977,36 @@ def ref_pairwise_distance(input1, input2):
         gradcheck_wrapper=gradcheck_wrapper_masked_operation
     ),
     OpInfo(
-        '_masked.softmax',
+        '_masked_softmax',
         method_variant=None,
+        supports_gradgrad=False,
         dtypes=floating_types_and(torch.bfloat16),
-        dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16),
-        sample_inputs_func=sample_inputs_masked_softmax,
+        dtypesIfCUDA=floating_types_and(torch.float16, *[torch.bfloat16] if CUDA11OrLater else []),
+        sample_inputs_func=sample_inputs_masked_softmax_v2,
         skips=(
             # torch.jit.frontend.NotSupportedError: Compiled
             # functions can't take variable number of arguments or
             # use keyword-only arguments with defaults
             DecorateInfo(unittest.skip("Skipped!"), 'TestJit', 'test_variant_consistency_jit'),
+            DecorateInfo(unittest.skip("Skipped!"), 'TestCommon', 'test_noncontiguous_samples'),
+            # torch.autograd.gradcheck.GradcheckError: Jacobian mismatch for output 0 with respect to input 0
+            DecorateInfo(unittest.expectedFailure, "TestGradients", "test_fn_grad", dtypes=(torch.float64,)),
```

Comment:
For CPU, I'm getting a failure in `TestGradients.test_fn_grad`. For a simple case, it's failing with the following comparison: 

```
Numerical:
 tensor([[0.]], dtype=torch.float64)
Analytical:
tensor([[nan]], dtype=torch.float64)
```

From a glance, it seems to be okay, because in small cases, the non-zero gradients seem to match, and the nan's correspond to 0.

I tried adding the `gradcheck_wrapper` to use `torch.where` to convert everything with a `True` mask to 0, but that still didn't work

Reply:
After this push, the CUDA implementation is completely resolved, but CPU is still running into this Numerical/Analytical difference. A bigger example of this: P476681114